app/controllers/concerns/params_require.rb

```diff
@@ -0,0 +1,37 @@
+# RubyGems.org gets a bunch of bad input from people testing our security.
+# This module provides a way to safely require params without crashing unnecessarily.
+#
+# Specifically it prevents a NoMethodError on permit for params.require(:scope).permit(:key) when
+# the user has tampered with the query params or request body to make params[:scope] a string.
+module ParamsRequire
+  # Safely require a hash parameter without crashing on bad input.
+  # params_require_hash(:user)
+  # => params.require(:user) if params[:user].respond_to?(:permit)
+  # raises ActionController::ParameterMissing if params[:user] is not a Hash
+  def params_require_hash(key, parameters = params)
+    value = parameters.require(key)
+    return value if value.respond_to?(:permit)
+    raise ActionController::ParameterMissing, key, parameters.keys
+  end
+
+  # Safely require an Arrayl parameter without crashing on bad input.
```

Comment:
```suggestion
  # Safely require an Array parameter without crashing on bad input.
```

Reply:
I ended up removing this whole module, so this is fixed.